01_explore_data/nbconverted/explore_cosmic_gene_set.py

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Download and explore COSMIC cancer genes
+# 
+# We want to download the set of cancer-associated genes from the [COSMIC Cancer Gene Census](https://cancer.sanger.ac.uk/cosmic/census), in order to use these genes in our experiments as a comparison/complement to the Vogelstein et al. gene set.
+
+# In[1]:
+
+
+from pathlib import Path
+
+import pandas as pd
+
+import mpmp.config as cfg
+import mpmp.utilities.data_utilities as du
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+
+# ### Load and process raw COSMIC data
+# 
+# This has to be downloaded directly from the CGC website linked above - you need an account there to download the .tsv file, so we can't do it programmatically.
+
+# In[2]:
+
+
+cosmic_df = pd.read_csv(
+    cfg.cosmic_genes_file, sep='\t', index_col=0
+)
+
+cosmic_df = cosmic_df[
+    # use only tier 1 genes
+    ((cosmic_df.Tier == 1) &
+    # drop genes without a catalogued somatic mutation
+     (cosmic_df.Somatic == 'yes') &
+    # drop genes that are only observed in cancer as fusions
+    # (we're not calling fusion genes in our mutation data)
+     (cosmic_df['Role in Cancer'] != 'fusion'))
+].copy()
+     
+print(cosmic_df.shape)
+cosmic_df.head()
+
+
+# ### Clean up the oncogene/TSG annotations
+# 
+# We want as many genes as possible to be annotated as _either_ an oncogene or TSG, so we know whether to use copy gain or copy loss data to define relevant CNV info. 
+# 
+# So, here, we will:
+# 
+# 1) drop genes that are annotated only as fusion genes (since we're not calling fusions at this time)  
+# 2) try to resolve genes that are annotated as both oncogene/TSG (usually context/cancer type specific) into their most likely pan-cancer category  
+# 3) for genes that can't be resolved confidently, we'll keep them as "oncogene, TSG" and run our scripts using both copy gain and copy loss downstream.
+
+# In[3]:
+
+
+print(cosmic_df['Role in Cancer'].unique())
+
+# if a gene is annotated as an oncogene/TSG and a fusion gene, just
+# get rid of the fusion component
+cosmic_df['Role in Cancer'] = cosmic_df['Role in Cancer'].str.replace(', fusion', '')
+print(cosmic_df['Role in Cancer'].unique())
+
+
+# In[5]:
+
+
+# load Bailey et al. data
+# supplementary table from https://www.sciencedirect.com/science/article/pii/S009286741830237X
+# this contains oncogene/TSG predictions for genes/cancer types using 20/20+ classifier
+class_df = pd.read_excel(
```

Comment:
Is there an efficient way to make sure that none of the genes got coerced into dates due to exposure to Excel?